src/devices/cpu/hcd62121/hcd62121.cpp

```diff
@@ -336,42 +383,240 @@ void hcd62121_cpu_device::device_start()
+	state_add(HCD62121_R1C, "R1C", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R20, "R20", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R24, "R24", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R28, "R28", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R2C, "R2C", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R30, "R30", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R34, "R34", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R38, "R38", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R3C, "R3C", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R40, "R40", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R44, "R44", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R48, "R48", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R4C, "R4C", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R50, "R50", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R54, "R54", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R58, "R58", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R5C, "R5C", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R60, "R60", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R64, "R64", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R68, "R68", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R6C, "R6C", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R70, "R70", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R74, "R74", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R78, "R78", m_debugger_temp).callimport().callexport().formatstr("%8s");
+	state_add(HCD62121_R7C, "R7C", m_debugger_temp).callimport().callexport().formatstr("%8s");
 
 	set_icountptr(m_icount);
 }
 
+void hcd62121_cpu_device::state_import(const device_state_entry &entry)
+{
+	switch (entry.index())
+	{
+		case HCD62121_R00:
+			m_reg[0x00] = (m_debugger_temp >> 24) & 0xff;
+			m_reg[0x01] = (m_debugger_temp >> 16) & 0xff;
+			m_reg[0x02] = (m_debugger_temp >> 8) & 0xff;
+			m_reg[0x03] = m_debugger_temp & 0xff;
+			break;
+		case HCD62121_R04:
+			m_reg[0x04] = (m_debugger_temp >> 24) & 0xff;
+			m_reg[0x05] = (m_debugger_temp >> 16) & 0xff;
+			m_reg[0x06] = (m_debugger_temp >> 8) & 0xff;
+			m_reg[0x07] = m_debugger_temp & 0xff;
+			break;
```

Comment:
Since you know these indices are sequential, and you know your register storage is linear, can’t you do something like:
```c++
...
#include "multibyte.h"
...
void hcd62121_cpu_device::state_import(const device_state_entry &entry)
{
	if ((entry.index() >= HCD62121_R00) && (entry.index() <= HCD62121_R7C))
	{
		put_u32be(&m_reg[(entry.index() - HCD62121_R00) * 4], m_debugger_temp);
	}
	else switch (entry.index())
	{
...
```